checkUrlExists/lambda/index.js

```diff
@@ -1,4 +1,18 @@
+const fs = require("fs");
+const tls = require("tls");
+
 const { curly } = require("node-libcurl");
+
+// important steps to get validation of https (as opposed to http) urls
+// https://stackoverflow.com/questions/63052127/protractor-node-libcurl-failed-ssl-peer-certificate-or-ssh-remote-key-was-not-o
+// Downloaded cacert.pem from https://curl.haxx.se/docs/caextract.html
```

Comment:
Confused by this comment -- you're not downloading the file (directly) that I can see; you're writing the root certs to /tmp down below on line 14.

Reply:
The file was downloaded by me before [building](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-build.html) the [SAM application](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html)

The AWS Serverless Application Model (AWS SAM) is an open-source framework that you can use to build serverless applications on AWS.

A serverless application is a combination of Lambda functions, event sources, and other resources that work together to perform tasks. Note that a serverless application is more than just a Lambda function—it can include additional resources such as APIs, databases, and event source mappings.

In this case, the SAM application includes the cacerts.pem file, the index.js and other node.js packages. 

The SAM application deployment package is uploaded to S3, and the location of the deployment package is provided as an input to the CF stack.

Reply:
> The file was downloaded by me before [building](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-build.html) the [SAM application](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html)
> 
> The AWS Serverless Application Model (AWS SAM) is an open-source framework that you can use to build serverless applications on AWS.
> 
> A serverless application is a combination of Lambda functions, event sources, and other resources that work together to perform tasks. Note that a serverless application is more than just a Lambda function—it can include additional resources such as APIs, databases, and event source mappings.
> 
> In this case, the SAM application includes the cacerts.pem file, the index.js and other node.js packages. In particular the template.yml file specifies the node.js version to use and where to find the index.js file when building the application.
> 
> The SAM application deployment package is uploaded to S3, and the location of the deployment package is provided as an input to the CF stack.

I sort of see (still grokking SAM). Where is this done? Should the build be in source control (what if we need to make another update to the SAM application beyond the cert)?

Reply:
Thanks for asking these questions!; I was mistaken on downloading the cacert.pem file. The root certs are retrieved via `tls.rootCertificates`; these are written to the file /tmp/cacerts.pem, and are used in the curly call. I've upated the comments in index.js.

The deployment build and package is done in CircleCI [here](https://github.com/dockstore/lambda/blob/188c27cd8b2f11e6072f271c28cd5be8ebdf5644/.circleci/config.yml#L99)

Reply:
RE: _Should the build be in source control (what if we need to make another update to the SAM application beyond the cert)_
I suppose if we update application components in a git branch, CircleCI will build those components and there will be a new application deployed...